WordPress/Classes/ViewRelated/Likes/LikesListController.swift

```diff
@@ -22,6 +22,7 @@ import WordPressKit
 
 class LikesListController: NSObject {
 
+    private let parent: UIViewController
```

Comment:
[Suggestion] if the view controller owns this `LikesListController`, which in turns owns the view controller through the parent variable, then this is a reference cycle and it will cause a memory leak. I'd suggest to mark `parent` as `weak`.

Then you can do the following when configuring the cells:

```swift
guard let parent else {
   return
}
cell.configure(..., parent: parent)
```

If the `parent` is nil, that means the view controller was deallocated and the screen is no longer visible to the user.

Reply:
Nice catch! Didn't notice the looped references.

Reply:
If it is ensured semantically that Child can only exist when Parent exists, you can also use `unowned` that removes the need to unwrap. It is kind of a forced unwrapped weak reference in practice. However it highlights the relationship between the 2 objects.

Reply:
Personally, I prefer using `weak` in all scenarios, as using `unowned` carries the risk of the app crashing if the child’s lifecycle outlives that of its parent

Reply:
Yeah, I researched the difference between `weak` and `unowned` modifiers, and I assume that `LikesListController` cannot naturally exist without the "parent" `NotificationDetailsViewController`. Such an incorrect state would lead to crashes and defects by default despite this value. I agree that using this modifier, in this case, more straightly indicates a cycled dependency, which is,  in my opinion, more beneficial.